### PR TITLE
movie, movieSession dto and controllers added, getById added to dao a…

### PR DESCRIPTION
**Pull request by DmGrt**
…nd service layers

### CODE REVIEW

**Review by margarita-ignatieva — APPROVED** [review]

File: src/main/java/com/spring/cinema/controllers/AuthenticationController.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import com.spring.cinema.models.dto.user.UserRegistrationDto;
+import com.spring.cinema.security.AuthenticationService;
+import org.apache.log4j.Logger;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping
```
Comment (Sofasmile) [inline src/main/java/com/spring/cinema/controllers/AuthenticationController.java]: do we need RequestMapping here?

File: src/main/java/com/spring/cinema/dao/impl/AbstractDao.java [hunk 2]
```diff
@@ -39,6 +39,17 @@ public T add(T item) {
         }
     }
 
+    public T getById(Long id, Class clazz) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<T> query = session.createQuery(
```
Comment (Sofasmile) [inline src/main/java/com/spring/cinema/dao/impl/AbstractDao.java]: 1 - https://mate-academy.github.io/jv-program-fulltime/04_spring/checklist/hw_33.html

File: src/main/java/com/spring/cinema/models/dto/movie/MovieResponseDto.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.spring.cinema.models.dto.movie;
+
+import lombok.Data;
+
+@Data
+public class MovieResponseDto {
+    private Long id;
+    private String title;
```
Comment (Sofasmile) [inline src/main/java/com/spring/cinema/models/dto/movie/MovieResponseDto.java:8]: let's add description too?

File: src/main/java/com/spring/cinema/models/dto/moviesession/MovieSessionDtoMapper.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import com.spring.cinema.service.MovieService;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MovieSessionDtoMapper {
+    private final MovieService movieService;
+
+    private CinemaHallService cinemaHallService;
```
Comment (Sofasmile) [inline src/main/java/com/spring/cinema/models/dto/moviesession/MovieSessionDtoMapper.java]: why do we use final with MovieService but not with CinemaHallService

File: src/main/java/com/spring/cinema/models/dto/user/UserRegistrationDto.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package com.spring.cinema.models.dto.user;
+
+import lombok.Data;
+
+@Data
+public class UserRegistrationDto {
```
Comment (Sofasmile) [inline src/main/java/com/spring/cinema/models/dto/user/UserRegistrationDto.java:6]:
```suggestion
public class UserRequestDto {
```

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**